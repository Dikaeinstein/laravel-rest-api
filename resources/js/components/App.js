import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Articles from '../components/Articles/ArticleList';
import SaveArticle from '../components/Articles/SaveArticle';
import UpdateArticle from '../components/Articles/UpdateArticle';
import ArticleDetail from '../components/Articles/ArticleDetail';
import Login from './Login';
import Register from './Register';
import Header from './Header';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Header />
          <Switch>
            <Route exact path='/' component={Login} />
            <Route exact path='/register' component={Register} />
            <Route exact path='/create' component={SaveArticle} />
            <Route exact path='/update' component={UpdateArticle} />
            <Route exact path='/articles' component={Articles} />
            <Route exact path='/articles/:id' component={ArticleDetail} />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

if (document.getElementById('root')) {
  ReactDOM.render(<App />, document.getElementById('root'));
}

export default App;
