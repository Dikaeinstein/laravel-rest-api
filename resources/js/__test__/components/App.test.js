import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';

describe('<App />', () => {
  it('should render correctly', () => {
    const tree = shallow(<App />);
    expect(tree).toMatchSnapshot();
  });
});
