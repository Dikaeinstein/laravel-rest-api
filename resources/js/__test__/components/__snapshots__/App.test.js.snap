// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<BrowserRouter>
  <div>
    <Header />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/register"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/create"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/update"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/articles"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/articles/:id"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
